jax/_src/numpy/polynomials/chebyshev.py

```diff
@@ -0,0 +1,605 @@
+        res = chebadd(chebmulx(res), pol[i])
+    return res[:-1]
+
+@jit
+def cheb2poly(c):
+    [c] = pu.as_series([c])
+    n = len(c)
+    if n < 3:
+        return c
+    else:
+        c0 = jnp.asarray(c[-2])
+        c1 = jnp.asarray(c[-1])
+
+        for i in range(n-1, 1, -1):
+            tmp = c0
+            c0 = chebsub(c[i-2], c1)
+            if len(c1.shape) == 0:
+                prd = jnp.empty(2, dtype=c1.dtype)
+            else:
+                prd = jnp.empty(len(c1)+1, dtype=c1.dtype)
+            prd = prd.at[0].set(0)
+            prd = prd.at[1:].set(c1)
+            c1 = chebadd(tmp, prd*2)
+        prd = jnp.empty(len(c1)+1, dtype=c1.dtype)
+        prd = prd.at[0].set(0)
+        prd = prd.at[1:].set(c1)
+        return chebadd(c0, prd)
+
+#Chebyshev default domain
+chebdomain = jnp.array([-1, 1])
+
+#Chebyshev coefficients representing zero
+chebzero = jnp.array([0])
+
+#Chebyshev coefficients representing one
+chebone = jnp.array([1])
+
+#Chebyshev coefficients representing the identity x
+chebx = jnp.array([0, 1])
+
+@partial(jit, static_argnames=('scl',))
+def chebline(off, scl):
+    if scl != 0:
+        return jnp.array([off, scl])
+    else:
+        return jnp.array([off])
+
+@jit
+def chebfromroots(roots):
+    if len(roots) == 0:
+        return jnp.ones(1)
+    else:
+        [roots] = pu.as_series([roots], trim=False)
+        roots.sort()
+        p = [chebline(-r, 1) for r in roots]
+        n = len(p)
+        while n > 1:
+            m, r = divmod(n, 2)
+            tmp = [chebmul(p[i], p[i+m]) for i in range(m)]
+            if r:
+                tmp[0] = chebmul(tmp[0], p[-1])
+            p = tmp
+            n = m
+        return p[0]
+
+@jit
+def chebadd(c1, c2):
+    [c1, c2] = pu.as_series([c1, c2])
+    c1, c2 = _promote_dtypes_inexact(c1, c2)
+    if len(c1) > len(c2):
+        c1 = c1.at[:c2.size].add(c2)
+        ret = c1
+    else:
+        c2 = c2.at[:c1.size].add(c1)
+        ret = c2
+    return ret
+
+@jit
+def chebsub(c1, c2):
+    [c1, c2] = pu.as_series([c1, c2])
+    c1, c2 = _promote_dtypes_inexact(c1, c2)
+    if len(c1) > len(c2):
+        c1 = c1.at[:c2.size].add(-c2)
+        ret = c1
+    else:
+        c2 = -c2
+        c2 = c2.at[:c1.size].add(c1)
+        ret = c2
+    return ret
+
+@jit
+def chebmulx(c):
+    [c] = pu.as_series([c])
+    c = _promote_dtypes_inexact(c)[0]  #Surely there is a better way to do this
+    prd = jnp.empty(len(c) + 1, dtype=c.dtype)
+    prd = prd.at[0].set(c[0]*0)
+    prd = prd.at[1].set(c[0])
+    tmp = c[1:]/2
+    prd = prd.at[2:].set(tmp)
+    prd = prd.at[0:-2].add(tmp)
+    return prd
+
+@partial(jit, static_argnames=('trim',))
+def chebmul(c1, c2, trim=False):
+    [c1, c2] = pu.as_series([c1, c2])
+    c1, c2 = _promote_dtypes_inexact(c1, c2)
+    z1 = _cseries_to_zseries(c1)
+    z2 = _cseries_to_zseries(c2)
+    prd = _zseries_mul(z1, z2)
+    ret = _zseries_to_cseries(prd)
+    if trim:
+        return pu.trimseq(ret)  #NO JIT
```

Comment:
What does `NO JIT` here mean? You've wrapped the outer function in `JIT` so this will be executed within a jit context.